tac-operation-lifecycle/tac-operation-lifecycle-logic/src/database.rs

```diff
@@ -1234,24 +1336,50 @@ impl TacDatabase {
 
     pub async fn get_operations(
         &self,
-        count: usize,
-        earlier_timestamp: Option<u64>,
-        sort: OrderDirection,
-    ) -> anyhow::Result<Vec<operation::Model>> {
+        pagination_input: Option<LogicPagination>,
+    ) -> anyhow::Result<(Vec<operation::Model>, Option<LogicPagination>)> {
         let mut query = operation::Entity::find();
 
-        if let Some(ts) = earlier_timestamp {
-            query = query.filter(Column::Timestamp.lt(Self::timestamp_to_naive(ts as i64)));
+        if let Some(pagination) = pagination_input {
+            query = query.filter(Column::Timestamp.lt(Self::timestamp_to_naive(
+                pagination.earlier_timestamp as i64,
+            )));
```

Comment:
this can be extracted as well

https://github.com/blockscout/blockscout-rs/blob/807f25e7a4343fcf85b225868e302b7817459ce2/blockscout-ens/bens-logic/src/subgraph/sql/utils.rs#L23 
seems pretty convenient to implement something similar

Reply:
Not as elegant as in `bens` service (since I work with `Select<>`, which is immutable, unlike `SelectStatement`). But here is my attempt https://github.com/blockscout/blockscout-rs/pull/1382/commits/0eb2e40edb32fa1953288b0bd9591f8fcf7e4384